xbmc/addons/Addon.h

```diff
@@ -258,54 +398,102 @@ class CAddon : public IAddon
   void OnPostUnInstall() override{};
 
 protected:
-  /*! \brief Whether or not the settings have been initialized. */
-  virtual bool SettingsInitialized() const;
+  /*!
+   * \brief Whether or not the settings have been initialized.
+   *
+   * \param[in] id Instance identifier to use, use @ref ADDON_SETTINGS_ID
+   *               to denote global add-on settings from settings.xml.
+   * \return true if settings initialize was successfull
+   */
+  virtual bool SettingsInitialized(AddonInstanceId id = ADDON_SETTINGS_ID) const;
 
-  /*! \brief Whether or not the settings have been loaded. */
-  virtual bool SettingsLoaded() const;
+  /*!
+   * \brief Whether or not the settings have been loaded.
+   *
+   * \param[in] id Instance identifier to use, use @ref ADDON_SETTINGS_ID
+   *               to denote global add-on settings from settings.xml.
+   * \return true if settings are loaded correct
+   */
+  virtual bool SettingsLoaded(AddonInstanceId id = ADDON_SETTINGS_ID) const;
 
-  /*! \brief Load the default settings and override these with any previously configured user settings
-   \param bForce force the load of settings even if they are already loaded (reload)
-   \param loadUserSettings whether or not to load user settings
-   \return true if settings exist, false otherwise
-   \sa LoadUserSettings, SaveSettings, HasSettings, HasUserSettings, GetSetting, UpdateSetting
+  /*!
+   * \brief Load the default settings and override these with any previously configured user settings
+   *
+   * \param[in] bForce force the load of settings even if they are already loaded (reload)
+   * \param[in] loadUserSettings whether or not to load user settings
+   * \param[in] id Instance identifier to use, use @ref ADDON_SETTINGS_ID
+   *               to denote global add-on settings from settings.xml.
+   * \return true if settings exist, false otherwise
+   *
+   * \sa LoadUserSettings, SaveSettings, HasSettings, HasUserSettings, GetSetting, UpdateSetting
    */
-  bool LoadSettings(bool bForce, bool loadUserSettings = true);
+  bool LoadSettings(bool bForce, bool loadUserSettings, AddonInstanceId id = ADDON_SETTINGS_ID);
```

Comment:
Think to add about here on another following request an change of bool to enum, like now a bit confusing and can be maybe by mistake become id instead of bool.

E.g.:
```cpp
enum class SettingsForceLoad
{
  YES,
  NO
};

enum class SettingsLoadUserPart
{
  YES,
  NO
};

bool LoadSettings(SettingsForceLoad bForce, SettingsLoadUserPart loadUserSettings, AddonInstanceId id = ADDON_SETTINGS_ID);
```

@ksooo What you think about?

Reply:
Better to avoid YES and NO as enum values as they seem to already exist in Darwin and have been a cause for past issues. @howie-f should know better (I think CHOICE_YES and CHOICE_NO has been used instead)

Reply:
Also relates to an "enum class"?

Reply:
Yes, see https://github.com/xbmc/xbmc/pull/20925

Reply:
@AlwinEsch generally I like the idea, but may I ask you not to add more stuff to this PR, please? It is already large and difficult to review. So, I would appreciate if you put this kind of cosmetic enhancements not needed for the addon-API changes in a follow-up PR. For now, let's focus on getting this PR in shape, without adding more to it.

Reply:
Yes, not thought about for here (see the "here on another following request" above), only as the place visible here I thought to ask about 😀.

Reply:
There is much more room for improving several function signatures, like getting rid of bool parameters, replacing them with speaking enum values etc. But all this is really for another PR.